File: backend/tests/fib_test.go
package fib

import (
	"testing"
)

func TestFib(t *testing.T) {
	if Fib(0) != 0 {
		t.Errorf("Erorr: fib(0) is not 0.")
	}

	if Fib(2) != 1 {
		t.Errorf("Erorr: fib(2) is not 2.")
	}

	if Fib(3) != 2 {
		t.Errorf("Erorr: fib(3) is not 3.")
	}

	if Fib(10) != 55 {
		t.Errorf("Erorr: fib(10) is not 55.")
	}

	if Fib(20) != 6765 {
		t.Errorf("Erorr: fib(20) is not 6765.")
	}

	if Fib(100) != 3736710778780434371 {
		t.Errorf("Erorr: fib(100) is not 3736710778780434371.")
	}

	if Fib(1234567) != 14656744304168315661 {
		t.Errorf("Erorr: fib(1234567) is not 14656744304168315661.")
	}
}

File: frontend/tests/fib.test.js

const fib = require('./fib').fib

test('testing the fibonacci function', () => {
	expect(fib(0)).toBe(0)
	expect(fib(2)).toBe(1)
	expect(fib(3)).toBe(2)
	expect(fib(10)).toBe(55)
	expect(fib(20)).toBe(6765)
	expect(fib(50)).toBe(12586269025)
})
